GoogleForCreators/web-stories-wp
# Dashboard: Responsive Typeahead UI Update

# Overview 
set up media queries and props to control typeahead component when on smaller viewports (800px and lower) that require size changes when interacting with typeahead

## Details
- Passing an `isExpanded` prop to relevant containers to control the look of the typeahead when viewport is 800px or smaller.
- Updating pixels to avoid pixels from old design agency, Sam's still updating the style guide so I suspect these change again but are 1 step closer to clean this way. 

## Testing 
- Pull the branch and run storybook, go to Dashboard/Components/TypeaheadInput 
- Use the storybook viewport button and swap over to 'iPad', see that the typeahead is now just the search icon. If you click on the icon the typeahead will grow. 
- If there's a value in the typeahead when you move away from the input it will stay expanded. 
- Clear the value and see the component return to just the icon view
- Now make your viewport larger and see that the typeahead is only ever expanded in view.

## Notes
- I am not sure if the typeahead style is to be controlled by viewport size or prop from the parent. From what I can tell, it's based on thew viewport according to figma. I think that the approach I took is easy enough to sub out as a prop if we need to though. 

![expanding typeahead](https://user-images.githubusercontent.com/10720454/77801580-6566c400-7036-11ea-8174-61a6fe54cc0f.gif)
